src/Common/Crypto/README.md

```diff
@@ -0,0 +1,264 @@
+
+## Overview
+
+The crypto strategy system is built around the [Strategy Pattern](https://refactoring.guru/design-patterns/strategy) with **install-time selection** to ensure data accessibility and prevent configuration drift.
+
+### Core Components
+
+- **`CryptoGen`** - Main encryption facade that loads strategy from database
+- **`EncryptionStrategyInterface`** - Contract that all strategies must implement (extends `Serializable`)
+- **`CryptoGenStrategy`** - Default AES-256-CBC + HMAC-SHA384 implementation
+- **`NullEncryptionStrategy`** - Identity function (no encryption)
+- **`EncryptionStrategySelector`** - Tool for strategy discovery and selection during installation
+- **`EncryptionStrategyRegistrationEvent`** - Event for modules to register available strategies
+- **`CryptoGenException`** - Exception for critical crypto errors
+
+## How the Strategy System Works
+
+### Install-Time Strategy Selection
+
+The encryption strategy is selected **once during installation** and stored in the database. This ensures:
+
+1. **Data accessibility** - Strategy remains available even if modules are removed
+2. **Configuration immutability** - No runtime strategy changes that could cause data loss
+3. **Consistency** - All encryption uses the same strategy throughout the system
+
+### Strategy Loading Priority
+
+When `CryptoGen` is instantiated, it loads the strategy in this order:
+
+1. **Database-stored strategy** (from installation)
+2. Default `CryptoGenStrategy` (fallback if no strategy stored)
+
+### Strategy Interface
+
+All encryption strategies must implement `EncryptionStrategyInterface` and `Serializable`:
+
+```php
+interface EncryptionStrategyInterface extends \Serializable
+{
+    public function encryptStandard(?string $value, ?string $customPassword = null, string $keySource = 'drive');
+
+    public function decryptStandard(?string $value, ?string $customPassword = null, string $keySource = 'drive', ?int $minimumVersion = null): false|string;
+
+    public function cryptCheckStandard(?string $value): bool;
+}
+```
+
+## Installation Integration
+
+The strategy system integrates with OpenEMR's installation process to allow strategy selection during setup.
+
+### Installation Flow
+
+1. **Strategy Discovery** - `EncryptionStrategySelector` dispatches `EncryptionStrategyRegistrationEvent::STRATEGY_REGISTRATION`
+2. **Module Registration** - Modules can register available strategies via event listeners
+3. **Strategy Selection** - User/installer selects strategy (default: `cryptogen`)
+4. **Serialization** - Selected strategy is serialized and stored in `globals.encryption_strategy_serialized`
+5. **Runtime Loading** - `CryptoGen` loads and deserializes strategy from database
+
+### Available Installation Options
+
+- **Web Installation** - Strategy selection integrated into installation UI
```

Comment:
I don't even think i would allow selection of this on the UI since those users have much less technical adeptness (if a user unknowingly chooses null then would be a complete disaster) and the autointaller via docker/orchestration is where this will almost always come into play. If needed, could support a get parameter via setup.php to open up the selector for edge cases.